### PR TITLE
Redesign

### PR DESCRIPTION
**Pull request by knownasilya**
Implementation of #400 mockups by @matmac using ember-paper

Don't merge yet, this is a WIP. Goal is to get it as close to the mockups as possible, but not deviate too much from Ember Paper semantics.

#### TODO
- [x] Move to sidebar layout
- [x] Integrate ember-paper
- [x] Replace tree with https://github.com/IvyApp/ivy-tree
  * [ ] Fix expand all/collapse all buttons (expand all possible with '*' key, but otherwise see https://github.com/IvyApp/ivy-tree/issues/11)
- [x] Hide sidebar on desktop
- [x] Finish file menu
  * [x] Add file menu items
  * [x] Add rename/move commands (good for someone trying to help)
  * [x] Move gist operations to top (out of file menu) (good for someone trying to help)
  * [x] Figure out certain nesting issues
- [x] Update styles (colors, icons, fonts)
- [x] Mobile support ~(flexi?)~
  * Not using Flexi atm due to no Ember 2.10 support
  * [x] Finish sidebar layout
  * [x] Finish main screen layout when sidebar closed
- [ ] Fix tests
- [ ] Fix bugs

#### Another PR
- Move file operation onto tree
- Alternate layouts support
- Save user settings (keymap mode, layout mode, etc) to gist or localstorage
- Monaco editor integration

Screenshot updated 1/15/17

screen shot 2017-01-15 at 1 50 31 pm

### CODE REVIEW

**Comment by knownasilya** [comment]
Here's an update:

twiddle-nav

**Comment by stefanpenner** [comment]
@knownasilya cool, I like the mobile responsive stuff!

**Comment by alexspeller** [comment]
I am super excited about this!

**Comment by Gaurav0** [comment]
@knownasilya Thanks so much for working on this!

Btw, I think you should merge master into your branch occasionally so that it doesn't become too difficult at some point.

**Comment by knownasilya** [comment]
@Gaurav0 thanks :) feel free to leave feedback!
I've been trying to rebase any chance I get.

**Comment by Gaurav0** [comment]
Build is failing on Travis.

**Comment by knownasilya** [comment]
@Gaurav0 should be fixed, not I only need to fix 43 tests, and more to come I'm sure.

**Comment by knownasilya** [comment]
Tree has been replaced with ivy-tree, see link above. Next step is to move some of the file operations directly onto the tree.

File: app/components/file-tree.js [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
-const { computed } = Ember;
+const { inject, computed } = Ember;
+const flatToNested = new FlatToNested();
 
 export default Ember.Component.extend({
+  treeService: inject.service('tree'),
   jsTreeActionReceiver: null,
 
-  fileTreeHash: computed('model.files.@each.filePath', function() {
+  fileTreeHash: computed('model.files.@ach.filePath', function() {
```
Comment (michalsnik) [inline app/components/file-tree.js]: I bet `@ach` is not intended here :)

**Comment by Gaurav0** [comment]
What are you using browserify for? I removed it with great difficulty for performance reasons.

**Comment by knownasilya** [comment]
I'm using it for https://github.com/ember-cli/ember-twiddle/pull/490/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R74 wasn't aware that it was removed since the browserify dep was still there. Thought it was an accidental removal. I'm up for alternatives though 👍

File: app/components/file-tree.js (hunk 1 above)
Comment (knownasilya) [inline app/components/file-tree.js]: Thanks!

**Comment by Gaurav0** [comment]
I can't see what you need it for. I am bringing in a browser build of babel-core via bower and using ember-cli-node-assets for things that don't need a browserified build, and replaced path with path-browser.

**Comment by Gaurav0** [comment]
I can't see what you need it for. I am bringing in a browser build of babel-core via bower and using ember-cli-node-assets for things that don't need a browserified build, and replaced path with path-browser. 

    On Thursday, February 2, 2017 12:35 PM, Ilya Radchenko <notifications@github.com> wrote:
 

 I'm using it for https://github.com/ember-cli/ember-twiddle/pull/490/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R74 wasn't aware that it was removed since the browserify dep was still there. Thought it was an accidental removal. I'm up for alternatives though 👍 —

**Comment by Gaurav0** [comment]
Ah, I see, you're using flat-to-nested. I recommend vendoring it.

**Comment by knownasilya** [comment]
Done, couldn't use nodeAssets, due to commonjs format, so just copied and added a define.

**Comment by ultimatemonty** [comment]
@knownasilya I'd be interested in helping here. I think this redesign will help with https://github.com/emberjs/website/issues/2369 - an issue we ran into was twiddle wasn't responsive enough to fit into the content size needed for the examples.

**Comment by knownasilya** [comment]
Feel free to start fixing tests, just submit PRs against this branch and I'll merge.

**Comment by knownasilya** [comment]
Started bringing back the file menu, and updated the tasks above. For anyone looking to help, check those.

File: bower.json [hunk 2]
```diff
@@ -12,8 +12,6 @@
     "lodash": "~3.10.1",
     "Faker": "~3.1.0",
     "jstree": "~3.3.0",
-    "hammer.js": "^2.0.8",
-    "matchMedia": "0.2.0",
```
Comment (knownasilya) [inline bower.json]: Ah, yes 👍

**Comment by knownasilya** [comment]
Todays update, started bringing over the gist save, fork, etc functionality

screen shot 2017-03-12 at 11 40 26 pm

File: app/components/gist-card.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+export default Ember.Component.extend({
+  session: inject.service(),
+  classNames: ['gist-card'],
+
+  // show fork option only if does not belong to user and is not a revision, otherwise show copy
+  // Github api does not permit forking if you own the gist already
+  // Github does not provide api for forking a revision
+  showFork: computed('gist.ownerLogin', 'session.currentUser.login', 'isRevision', function() {
+    return !this.get('isRevision') && this.get('gist.ownerLogin') !== this.get('session.currentUser.login');
+  }),
```
Comment (knownasilya) [inline app/components/gist-card.js]: Could you explain?

**Comment by sly7-7** [comment]
@knownasilya Is this still a thing you're working on ?

**Comment by knownasilya** [comment]
Yeah, whenever I get a chance. Would love help in the test area.

**Comment by lukesargeant** [comment]
@knownasilya I'm happy to try to help get this past the finish line but I've been having issues running the project locally. See #582

**Comment by jrock2004** [comment]
Going to start working on test fixes and I will also merge in code from master fork

**Comment by gokatz** [comment]
Hey @knownasilya, really excited to see the new design of Twiddle. Thank you so much for your time 😃 One small clarification, this PR has over 100 file changes. Won't it become tedious to review as a whole?

**Comment by knownasilya** [comment]
@gokatz yeah, probably. Reviews weren't even a thing when I started this lol. I might pull it out into smaller PRs

**Comment by gokatz** [comment]
Awesome 🎉 @knownasilya

**Comment by Gaurav0** [comment]
Closing as stale.

**Pull request closed without merging**